From e2a4c63807da5fc994dbf41b828d99f069d888f7 Mon Sep 17 00:00:00 2001
From: Ramasamy Kaliappan <quic_rkaliapp@quicinc.com>
Date: Fri, 10 Nov 2023 13:13:36 +0530
Subject: [PATCH] wpa_supplicant: fix to read feq_list from config

When enables 3Link mld station, it adds channel lists for all 3Link bands
in the freq list in wpa_supplicant configuration. while reading
freq list from wpa_supplicant config, it getting failed due to
insufficient buf size.

To fix this issue increase sufficient buf size in wpa_config_read.

Signed-off-by: Ramasamy Kaliappan <quic_rkaliapp@quicinc.com>
---
 wpa_supplicant/config_file.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/wpa_supplicant/config_file.c
+++ b/wpa_supplicant/config_file.c
@@ -303,7 +303,7 @@ struct wpa_config * wpa_config_read(cons
 				    bool ro)
 {
 	FILE *f;
-	char buf[512], *pos;
+	char buf[1024], *pos;
 	int errors = 0, line = 0;
 	struct wpa_ssid *ssid, *tail, *head;
 	struct wpa_cred *cred, *cred_tail, *cred_head;
